rust/operator-binary/src/pod_svc_controller.rs

```diff
@@ -47,21 +58,28 @@ pub struct Ctx {
 
 const APP_KUBERNETES_LABEL_BASE: &str = "app.kubernetes.io/";
 
-pub async fn reconcile_pod(pod: Arc<Pod>, ctx: Arc<Ctx>) -> Result<Action, Error> {
+pub async fn reconcile_pod(pod: Arc<Pod>, ctx: Arc<Ctx>) -> Result<Action> {
     tracing::info!("Starting reconcile");
 
     let name = pod.metadata.name.clone().context(PodHasNoNameSnafu)?;
 
-    let pod_labels = pod
-        .metadata
-        .labels
-        .as_ref()
-        .with_context(|| PodHasNoLabelsSnafu { name: name.clone() })?;
+    let pod_labels = Labels::try_from(
+        pod.metadata
+            .labels
+            .as_ref()
+            .with_context(|| PodHasNoLabelsSnafu { name: name.clone() })?,
```

Comment:
```suggestion
            .as_ref_or_else(BTreeMap::default),
```

Then again, this is all going away soon anyway.

Reply:
Nice, yeah we had wondered why it would error out rather than just _not have any labels_.

Reply:
Hmm, I can't seem to get that working. Maybe I need to jump on a call with you.

Reply:
What error do you get?

Reply:
If I `use stackable_operator::utils::OptionExt;`

```
no method named `as_ref_or_else` found for enum `std::option::Option` in the current scope
items from traits can only be used if the trait is in scope
```


and if I `use stackable_operator::utils::OptionExt as _;`

```
the trait bound `stackable_operator::kvp::Labels: From<Cow<'_, BTreeMap<std::string::String, std::string::String>>>` is not satisfied
the following other types implement trait `TryFrom<T>`:
  <stackable_operator::kvp::Labels as TryFrom<BTreeMap<K, V>>>
  <stackable_operator::kvp::Labels as TryFrom<[(K, V); N]>>
  <stackable_operator::kvp::Labels as TryFrom<&BTreeMap<K, V>>>
required for `Cow<'_, BTreeMap<std::string::String, std::string::String>>` to implement `Into<stackable_operator::kvp::Labels>`
required for `stackable_operator::kvp::Labels` to implement `TryFrom<Cow<'_, BTreeMap<std::string::String, std::string::String>>>`
```

and the rest of the code starts showing a lot of errors because of inferred types. Probably easier to share my screen.

Reply:
@nightkr I was having a look through this code a bit closer.
Currently we error out if there are no labels found on the pod.

You are suggesting to not throw and error, and simply return an empty set of labels. Sounds reasonable at first glance.

But these labels are then filtered (for `app.kubernetes.io/*`) and used as labels for a service.

- [ ] Should we continue to error out as was already done?
- [ ] Return an empty set of labels so the service ends up being created without any of the pods `app.kubernetes.io/*` labels?

I would prefer the first option for this PR.

Reply:
Ah ok, sorry I thought they were used as a selector, but they are just used as labels on the service. So not too dangerous, but maybe we want to handle a weird state where the pod has no labels, and then also if there are no `app.kubernetes.io/*` labels?

Reply:
I wouldn't say that's the responsibility of the pod svc controller to handle.

Reply:
So, @Techassi how can I do `.as_ref_or_else(BTreeMap::default)` but for `Labels`?

I've tried `Labels::default`, `Labels::new`. The error is posted a few comments above this.

Reply:
Okay, let's not block on this. It's not a regression and will die soon in #450 anyway.